blog/views.py

```diff
@@ -0,0 +1,60 @@
+from django.core.paginator import Paginator
+from django.db.models import Count
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import render, redirect, get_object_or_404
+
+from blog.forms import CommentaryForm
+from blog.models import User, Post, Commentary
+
+
+def index(request: HttpRequest) -> HttpResponse:
+    user_count = User.objects.count()
+    post_count = Post.objects.count()
+    comment_count = Commentary.objects.count()
+    post_list = Post.objects.annotate(
+        comments_count=Count("comments")
+    ).order_by("-created_time")
+
+    paginator = Paginator(post_list, 5)
+    page_number = request.GET.get("page")
+    page_obj = paginator.get_page(page_number)
+
+    context = {
+        "user_count": user_count,
+        "post_count": post_count,
+        "comment_count": comment_count,
+        "post_list": page_obj.object_list,
+        "paginator": paginator,
+        "is_paginated": paginator.num_pages > 1,
+        "page_obj": page_obj,
+    }
+
+    return render(request, "blog/index.html", context=context)
```

Comment:
It would be much easier to implement this through `generic`. Here we can use `ListView`, because we just need to display a list of objects (posts).

The task didn't ask to show the number of comments, etc., so we don't need to add it

> Inside blog.views create a view for the index url. This view returns a list of all posts, in descending order by created_time.